src/openrct2/peep/peep.c

```diff
@@ -1298,25 +1300,28 @@ static void sub_68F41A(rct_peep *peep, sint32 index)
     }
 
     uint8 energy = peep->energy;
-    uint8 energy_growth = peep->energy_target;
-    if (energy >= energy_growth){
+    uint8 energy_target = peep->energy_target;
+    if (energy >= energy_target)
+    {
         energy -= 2;
-        if (energy < energy_growth)
-            energy = energy_growth;
+        if (energy < energy_target)
+            energy = energy_target;
     }
-    else{
-        energy = min(255, energy + 4);
-        if (energy > energy_growth)
-            energy = energy_growth;
+    else
+    {
+        energy = min(PEEP_MAX_ENERGY_TARGET, energy + 4);
```

Comment:
This doesn't look correct. It will always end up being `energy = energy + 4`, because the only way it could end up as `PEEP_MAX_ENERGY_TARGET` is if `energy + 4 >= 256`, but that's not a valid value for `uint8`.

Reply:
It _could_ have been like that in the original.

Reply:
That doesn't make it any more correct, though.

Reply:
Perhaps an add_clamp_uint8 would be more appropriate more then?

Reply:
```
template<class T, class U> int8 __CFADD__(T x, U y)
{
  int size = sizeof(T) > sizeof(U) ? sizeof(T) : sizeof(U);
  if ( size == 1 )
    return uint8(x) > uint8(x+y);
  if ( size == 2 )
    return uint16(x) > uint16(x+y);
  if ( size == 4 )
    return uint32(x) > uint32(x+y);
  return uint64(x) > uint64(x+y);
}
 
{
  
  if ( peep->energy < peep->energy_target )
  {
    v6 = __CFADD__(peep->energy, 4);
    peep->energy = peep->energy + 4;
    if ( v6 )
      peep->energy = -1;
    if ( peep->energy > peep->energy_target )
      peep->energy = peep->energy_target;
  }
  else
  {
    peep->energy = peep->energy - 2;
    if ( peep->energy < peep->energy_target )
      peep->energy = peep->energy_target;
  }
  
  if ( peep->energy < 0x20u )
    peep->energy = 0x20u; // Min energy
  if ( peep->energy > 0x80u )
    peep->energy = 0x80u; // Max energy
 
}
```
Thats how the original game does it, keep in mind that __CFADD__ is something from IDA to emulate carry add as seen at rct2.exe:0x0068FB9E

The current version looks slightly different.